micropip/_compat_in_pyodide.py

```diff
@@ -21,6 +22,27 @@
         raise
     # Otherwise, this is pytest test collection so let it go.
 
+if IN_BROWSER:
+
+    async def _pyfetch(urls: str, **kwargs):
+        if "signal" in kwargs:
+            return await pyfetch(urls, **kwargs)
+
+        controller = AbortController.new()
+        kwargs["signal"] = controller.signal
+
+        async def fetch_with_abort():
+            try:
+                return await pyfetch(urls, **kwargs)
+            except CancelledError:
+                controller.abort()
+                raise
```

Comment:
Move this into a function taking controller, urls, kwargs as arguments instead of a closure?

Reply:
This is done. and `urls` is a typo so I [amend the commit](https://github.com/pyodide/micropip/compare/703fc54f71c0d666a10e7aeb0353f1364f56aca7..e01491f5051e8b03752563f18ddd425644e6ecc3)